common/src/main/java/io/quarkiverse/tekton/common/utils/Params.java

```diff
@@ -59,10 +59,21 @@ public static List<Param> create(List<String> list) {
     }
 
     public static Param create(String name, Object value) {
-        String stringVal = value instanceof String ? (String) value : null;
+        String stringVal = null;
         String[] arrayVal = value instanceof String[] ? (String[]) value : null;
         Map<String, String> objectVal = value instanceof Map ? (Map) value : null;
 
+        if (value instanceof String) {
```

Comment:
There could be edge cases where a param contains string enclosed in single or double quotes where space could be allowed inside them.

As we don't generate such params ourselves (yet), we can let it as is, but its something that we need to keep in mind.

Reply:
We could review that later if such a use case occurs and create a new PR.

Reply:
Added TODO part of the code